src/markdown/tutorial/part-2/11-ember-data.md

```diff
@@ -251,60 +253,95 @@ Darn, there were a couple of failing tests! At the same time, it's great that we
-* When performing the `find('rental', 'grand-old-mansion')` query, it requested the data from `/rentals/grand-old-mansion`, instead of `/api/rentals/grand-old-mansion.json`.
+- When building the `query('rentals')` request, the resulted `url` in request options was `/rentals`, instead of `/api/rentals.json`.
+- When building the `findRecord('rentals', 'grand-old-mansion')` request, the resulted `url` in request options was `/rentals/grand-old-mansion`, instead of `/api/rentals/grand-old-mansion.json`.
 
 Hm, okay, so we have to teach EmberData to fetch data from the correct location. But how does EmberData know how to fetch data from our server in the first place?
 
-## Working with Adapters and Serializers
+## Working with Request builders and Handlers
 
-EmberData uses an *[adapter](../../../models/customizing-adapters/)* and *[serializer](../../../models/customizing-serializers/)* architecture. Adapters deal with *how* and *where* EmberData should fetch data from your servers, such as whether to use HTTP, HTTPS, WebSockets or local storage, as well as the URLs, headers and parameters to use for these requests. On the other hand, serializers are in charge of converting the data returned by the server into a format EmberData can understand.
+Let's start customizing the things that didn't work for us by default. Specifically:
 
-The idea is that, provided that your backend exposes a *consistent* protocol and interchange format to access its data, we can write a single adapter-serializer pair to handle all data fetches for the entire application.
+- Our resource URLs have an extra `/api` *namespace* prefix.
+- Our resource URLs have a `.json` extension at the end.
 
-As it turns out, JSON:API just happens to be EmberData's default data protocol and interchange format. Out of the box, EmberData provides a default JSON:API adapter and serializer. This is great news for us, since that is also what our server has implemented. What a wonderful coincidence!
+The first thing we want to do is have our builder respect a configurable default host and/or namespace. Adding a namespace prefix happens to be pretty common across Ember apps, so EmberData provides a global config mechanism for host and namespace. Typically you will want to do this either in your store file or app file.
 
-However, as mentioned above, there are some minor differences between how our server works and EmberData's default assumptions. We can customize the default behavior by defining our own adapter and serializer:
+```run:file:patch lang=js cwd=super-rentals filename=app/app.js
+@@ -2,6 +2,11 @@
+ import Application from '@ember/application';
+ import Resolver from 'ember-resolver';
+ import loadInitializers from 'ember-load-initializers';
+ import config from 'super-rentals/config/environment';
++import { setBuildURLConfig } from '@ember-data/request-utils';
++
++setBuildURLConfig({
++  namespace: 'api',
```

Comment:
is it possible to have multi-api setup where namespaces is different per resource?

Reply:
Probably with a handler.

Reply:
It doesn't make sense to do per-resource because resources rarely-if-ever map cleanly to endpoints and formats.

However, if you wanted to do this, the solution is the same as how you would handle multiple endpoints and formats more generally: each gets their own builders and handlers. This includes that the host, namespace and resourcePath are overridable to any of the default builders on a per-request basis, though if you have a lot of such juggling to do its better to abstract it into separate builders targeting each spec.

Reply:
I think if you really have big multi-api setup (like multiple models/schemas per API) it might be better to have multiple stores configured. 
But simplest would be just have different set of builders